pkg/sessionctx/variable/sysvar.go

```diff
@@ -2965,6 +2966,41 @@ var defaultSysVars = []*SysVar{
 		}},
 }
 
+// GlobalSystemVariableInitialValue gets the default value for a system variable including ones that are dynamically set (e.g. based on the store)
+func GlobalSystemVariableInitialValue(varName, varVal string) string {
+	switch varName {
+	case TiDBTxnMode:
+		if config.GetGlobalConfig().Store == "tikv" || config.GetGlobalConfig().Store == "unistore" {
+			varVal = "pessimistic"
+		}
+	case TiDBEnableAsyncCommit, TiDBEnable1PC:
+		if config.GetGlobalConfig().Store == "tikv" {
+			varVal = On
```

Comment:
Can we retrieve the default value from a centralized location instead of hard coding it here? This could be a problem if we ever need to change the default value in the future but forget to modify this here.

Reply:
```
diff --git a/pkg/sessionctx/variable/sysvar.go b/pkg/sessionctx/variable/sysvar.go
index 532cd0f..e74cdf9 100644
--- a/pkg/sessionctx/variable/sysvar.go
+++ b/pkg/sessionctx/variable/sysvar.go
@@ -2093,7 +2093,7 @@ var defaultSysVars = []*SysVar{
                s.ShardAllocateStep = TidbOptInt64(val, DefTiDBShardAllocateStep)
                return nil
        }},
-       {Scope: ScopeGlobal | ScopeSession, Name: TiDBEnableAsyncCommit, Value: BoolToOnOff(DefTiDBEnableAsyncCommit), Type: TypeBool, 
+       {Scope: ScopeGlobal | ScopeSession, Name: TiDBEnableAsyncCommit, Value: defaultTiDBEnableAsyncCommit(), Type: TypeBool, SetSess
                s.EnableAsyncCommit = TiDBOptOn(val)
                return nil
        }},
@@ -2965,6 +2965,13 @@ var defaultSysVars = []*SysVar{
                }},
 }
 
+func defaultTiDBEnableAsyncCommit() string {
+       if config.GetGlobalConfig().Store == "tikv" {
+               return On
+       }
+       return BoolToOnOff(DefTiDBEnableAsyncCommit)
+}
+
```

I tried this way. But it seems `defaultSysVars ` is inited early than `config.GetGlobalConfig()`, so it can't work.
Maybe we can set `variable.GetSysVars` after `config.GetGlobalConfig()` loaded, but it is also "hard coding".

Reply:
* if we need set diffrent defalut value for diffrent store types for example `TiDBEnableAsyncCommit`，we need do that after `config.GetGlobalConfig()` loaded
* others，we can move into `defaultSysVars`, for example `TiDBRowFormatVersion`.

/cc @JmPotato

Reply:
It looks like this PR has minimal changes to fix the bug. If we want a more elegant way to make it, some questions should be answered before:

1. Do we need to keep the "right" default value for unistore which is only used for testing ? If we do not need it, we can make variables like `TiDBEnableAsyncCommit` configured statically without reading global config.
2. If cluster upgrades from an old one. What is the right default value for the variables that don't want to change. Just like `tidb_pessimistic_txn_fair_locking`, it should keep the old default value 'OFF' after upgrading and 'ON' for a new cluster.